tests/scripts/thread-cert/Cert_5_1_01_RouterAttach.py

```diff
@@ -168,6 +179,317 @@ def test(self):
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {\
+                              LEADER_DATA_TLV,\
+                              ROUTE64_TLV,\
+                              SOURCE_ADDRESS_TLV\
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255).\
+            must_next()
+
+        # Step 2: Router sends a MLE Parent Request with an IP hop limit of
+        #         255 to the Link-Local All Routers multicast address (FF02::2).
+        #         The following TLVs MUST be present in the MLE Parent Request:
+        #            - Challenge TLV
+        #            - Mode TLV
+        #            - Scan Mask TLV
+        #                If the DUT sends multiple MLE Parent Requests
+        #                    - The first one MUST be sent only to all Routers
+        #                    - Subsequent ones MAY be sent to all Routers and REEDS
+        #            -  Version TLV
+        #         If the first MLE Parent Request was sent to all Routers and
+        #         REEDS, the test fails.
+
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {\
+                              CHALLENGE_TLV,\
+                              MODE_TLV,\
+                              SCAN_MASK_TLV,\
+                              VERSION_TLV\
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255 and\
+                   p.mle.tlv.scan_mask.r == 1 and\
+                   p.mle.tlv.scan_mask.e == 0).\
+            must_next()
+        pkts_copy = pkts.copy()
+        start_index = pkts.index
+
+        # Step 3: Leader responds with a MLE Parent Response.
+        #         The following TLVs MUST be present in the MLE Parent Response:
+        #             - Challenge TLV
+        #             - Connectivity TLV
+        #             - Leader Data TLV
+        #             - Link-layer Frame Counter TLV
+        #             - Link Margin TLV
+        #             - Response TLV
+        #             - Source Address
+        #             - Version TLV
+        #             - MLE Frame Counter TLV (optional)
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_dst64(ROUTER).\
+            filter_mle_cmd(MLE_PARENT_RESPONSE).\
+            filter(lambda p: {\
+                              CHALLENGE_TLV,\
+                              CONNECTIVITY_TLV,\
+                              LEADER_DATA_TLV,\
+                              LINK_LAYER_FRAME_COUNTER_TLV,\
+                              LINK_MARGIN_TLV,\
+                              RESPONSE_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV,\
+                              MLE_FRAME_COUNTER_TLV\
```

Comment:
```suggestion
            filter(lambda p: {
                              CHALLENGE_TLV,
                              CONNECTIVITY_TLV,
                              LEADER_DATA_TLV,
                              LINK_LAYER_FRAME_COUNTER_TLV,
                              LINK_MARGIN_TLV,
                              RESPONSE_TLV,
                              SOURCE_ADDRESS_TLV,
                              VERSION_TLV,
                              MLE_FRAME_COUNTER_TLV
```
not necessary to use `\` ?